src/libraries/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -291,6 +291,32 @@ public unsafe void ToLogFont(object logFont, Graphics graphics)
             }
         }
 
+#if NET7_0_OR_GREATER
+        /// <summary>
+        /// Update the given LOGFONT with data from given graphics.
+        /// </summary>
+        /// <param name="logFont">A structure to populate with LOGFONT data.</param>
+        /// <param name="graphics">Graphics object which provide information about font.</param>
+        public unsafe void ToLogFont<T>(ref T logFont, Graphics graphics)
+            where T: struct
+        {
+            ArgumentNullException.ThrowIfNull(logFont);
+
+            int nativeSize = sizeof(Interop.User32.LOGFONT);
+            if (Marshal.SizeOf<T>() != nativeSize)
+            {
+                // If we don't actually have an object that is LOGFONT in size, trying to pass
+                // it to GDI+ is likely to cause an AV.
+                throw new ArgumentException(null, nameof(logFont));
+            }
+
+            Interop.User32.LOGFONT nativeLogFont = ToLogFontInternal(graphics);
+
+            // PtrToStructure requires that the passed in object not be a value type.
+            Marshal.PtrToStructure<T>(new IntPtr(&nativeLogFont), logFont);
```

Comment:
It would be best if these new APIs did not depend on the built-in runtime marshalling support at all.

Can these new APIs take `void*` that points to `LOGFONT` and require the callers to take of the marshalling if there is any marshalling required?

Reply:
Would change like this `logFont = *(T*)&nativeLogFont;` works?

Reply:
I think I get why you propose `void*`, because any struct which given to method can have managed fields, then I cannot copy data over layout. @RussKie can we just adopt these small methods to WinForms in appropriate places, having new `void*`  methods in `System.Drawing.Common` are probably not worth goal to pursue

Reply:
> can we just adopt these small methods to WinForms in appropriate places, having new `void*` methods in `System.Drawing.Common` are probably not worth goal to pursue

We can surely consider that.

Reply:
Looping in @JeremyKuhne as he may have thoughts on the subject as well.

Reply:
Have we considered the `unmanaged` constraint? Then you can do everything without `Marshal`. @jkotas?

Reply:
`unmanaged` constraint works for me.